core/src/nucleus/actions/initialize.rs

```diff
@@ -0,0 +1,113 @@
+extern crate futures;
+use action::{Action, ActionWrapper};
+use agent::actions::commit::commit_entry;
+use context::Context;
+use futures::{future, Async, Future};
+use hash_table::sys_entry::ToEntry;
+use holochain_dna::Dna;
+use instance::dispatch_action_and_wait;
+use nucleus::{
+    ribosome::callback::{genesis::genesis, CallbackParams, CallbackResult},
+    state::NucleusStatus,
+};
+use std::{sync::Arc, thread};
+
+/// Initialize Application, Action Creator
+/// This is the high-level initialization function that wraps the whole process of initializing an
+/// instance. It creates both InitApplication and ReturnInitializationResult actions asynchronously.
+///
+/// Returns a future that resolves to an Ok(NucleusStatus) or an Err(String) which carries either
+/// the Dna error or errors from the genesis callback.
+///
+/// Use futures::executor::block_on to wait for an initialized instance.
+pub fn initialize_application(
+    dna: Dna,
+    context: Arc<Context>,
+) -> Box<dyn Future<Item = NucleusStatus, Error = String>> {
+    if context.state().unwrap().nucleus().status != NucleusStatus::New {
```

Comment:
My regular question in these reviews hehe. But will this panic? is it meant to panic?

Reply:
If this function was called with an uninitialized context, i.e. one that does not have a state, I'd say yes that should panic since it's not a user error but something that should bother a dev to fix it.

Reply:
That works for me and makes a lot of sense. Nice to see panic being unsed in the context library design.